packages/nextjs/app/debug/_components/contract/utilsDisplay.tsx

```diff
@@ -40,11 +46,20 @@ export const displayTxResult = (
     }
 
     const type = functionOutputs[0].type;
-    const parsedParam = parseParamWithType(type, displayContent, true);
+    const parsedParam = displayContent as any;
 
     if (typeof parsedParam === "object")
       return JSON.stringify(parsedParam, replacer);
 
+    if (isCairoContractAddress(type)) {
+      return asText ? (
+        // BigNumberish here if you look at the code, the tostring method does not work properly
+        getChecksumAddress(`0x${parsedParam.toString(16)}`)
+      ) : (
+        <Address address={`0x${parsedParam.toString(16)}` as `0x${string}`} />
+      );
+    }
+
```

Comment:
Relevant changes that fix #319 , issue was caused by improper type conversion for the address.